src/Generator/Generators/CSharp/CSharpTypePrinter.cs

```diff
@@ -686,6 +686,31 @@ private static string GetParameterUsage(ParameterUsage usage)
         }
 
         private CSharpExpressionPrinter expressionPrinter => new CSharpExpressionPrinter(this);
+
+        public override TypePrinterResult VisitFieldDecl(Field field)
+        {
+            List<TranslationUnit> unit = new List<TranslationUnit>();
+            var cs = new CSharpSources(Context, unit);
+            var safeIdentifier = cs.SafeIdentifier(field.Name);
+
+            PushMarshalKind(MarshalKind.NativeField);
+            var fieldTypePrinted = field.QualifiedType.CSharpType(this);
```

Comment:
If you debug CSharpType you'll see it's a complicated wrapper equivalent to just Visit(this) so please use that instead.